token/program/src/processor.rs

```diff
@@ -67,6 +80,7 @@ impl Processor {
         if *mint_info.key == crate::native_mint::id() {
             account.is_native = true;
             account.amount = new_account_info.lamports();
```

Comment:
I think we want to exclude `rent_exempt_reserve` from the account amount.  Generically a client is going to look at the `amount` field to determine the max amount that can be transferred.  Since the `rent_exempt_reserve` can't be transferred, if not then the client is going to need special casing for wrapped SOL